contentcuration/contentcuration/assets/channelList/vuex/channelList/mutations.js

```diff
@@ -0,0 +1,73 @@
+import Vue from 'vue';
+import { prepChannel, setDirtyTracking } from '../../utils';
+
+/* CHANNEL LIST MUTATIONS */
+function mergeChannel(channelsMap, channel) {
+  setDirtyTracking(channel);
+  return {
+    ...channelsMap,
+    [channel.id]: {
+      ...channelsMap[channel.id],
+      ...channel,
+    },
+  };
+}
+
+export function ADD_CHANNEL(state, channel) {
+  state.channelsMap = mergeChannel(state.channelsMap, channel);
+}
+
+export function ADD_CHANNELS(state, channels = []) {
+  state.channelsMap = channels.reduce((channelsMap, channel) => {
+    return mergeChannel(channelsMap, channel);
+  }, state.channelsMap);
+}
+
+export function REMOVE_CHANNEL(state, channelId) {
+  Vue.delete(state.channelsMap, channelId);
+}
+
+export function UPDATE_CHANNEL(
+  state,
+  { id, name = null, description = null, thumbnailData = null, language = null } = {}
```

Comment:
This seemed simpler than creating a mutation for every different attribute, and goes some way to describe the channel schema I was talking about above.